addon/components/document-grid.js

```diff
@@ -1,11 +1,17 @@
 import { action } from "@ember/object";
 import { inject as service } from "@ember/service";
 import Component from "@glimmer/component";
```

Comment:
```suggestion
import UploadComponent from "ember-alexandria/components/upload";
```

Reply:
Add a new component file where you put in the upload task from the DocumentUploadButton. Then extend both the upload button and the grid component from it.

Reply:
Isn't a component that doesn't have a template a rather hacky approach simulating what mixins did?

Wouldn't it be cleaner to move this kind of work to a service?

Reply:
Fair. It doesnt have to be a Component can also be something like `utils/file-upload-task.js` which exposes a class with the upload task. Im personally a bit hesitant about putting all api actions in services. Il have to research this pattern a bit to find out more about it.

Reply:
I asked in our team-chat about the service-thing. It seems like the ember community uses this pattern.

Reply:
So. Service is ok?